### PR TITLE
refactor(storage-transfer): convert ava tests to mocha

### PR DESCRIPTION
**Pull request by fhinkel**
See failing tests in https://github.com/GoogleCloudPlatform/nodejs-docs-samples/pull/1138

### CODE REVIEW

**Comment by fhinkel** [comment]
I'm having some trouble with these tests being flaky: 

```
  1 passing (4m)
  4 failing
  1) should create a storage transfer job:
     Error: Timeout of 60000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/tmpfs/src/github/nodejs-docs-samples/storage-transfer/system-test/transfer.test.js)
  2) should get a transferJob:
     Error: Timeout of 60000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/tmpfs/src/github/nodejs-docs-samples/storage-transfer/system-test/transfer.test.js)
  3) should update a transferJob:
     Error: Timeout of 60000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/tmpfs/src/github/nodejs-docs-samples/storage-transfer/system-test/transfer.test.js)
  4) should list transferJobs:
     Error: Timeout of 60000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/tmpfs/src/github/nodejs-docs-samples/storage-transfer/system-test/transfer.test.js)
```

Adding `setTimeout(done, 2000)` back in.

File: storage-transfer/package.json [hunk 1]
```diff
@@ -12,9 +12,9 @@
     "node": ">=8"
   },
   "scripts": {
-    "unit-test": "repo-tools test run --cmd ava -- -T 20s --verbose test/*.test.js",
-    "system-test": "repo-tools test run --cmd ava -- -T 20s --verbose system-test/*.test.js",
-    "test": "npm run unit-test && npm run system-test"
+    "unit-test": "mocha test/*.test.js --timeout=10000",
```
Comment (JustinBeckwith) [inline storage-transfer/package.json]: So no need to block on this feedback, but I would rather just use `mocha --timeout=10000` across the board here.  That way if a new test gets added to the `test` directory, you don't have to worry about it having the right prefix.

File: storage-transfer/system-test/transfer.test.js [hunk 2]
```diff
@@ -16,49 +16,46 @@
...
-const status = `DISABLED`;
+const date = '2222/08/11';
+const time = '15:30';
+const description = 'this is a test';
+const status = 'DISABLED';
 
-test.before(tools.checkCredentials);
-test.before(async () => {
+before(async () => {
+  tools.checkCredentials();
```
Comment (JustinBeckwith) [inline storage-transfer/system-test/transfer.test.js]: We have just dropped that line from all the samples, and the console stubbing.  It's fine. Should we drop it here too?

File: storage-transfer/system-test/transfer.test.js [hunk 3]
```diff
@@ -16,49 +16,46 @@
...
-  });
+  const [bucket1] = await storage.createBucket(firstBucketName);
+  await bucket1.acl.add(bucketOptions);
+  const [bucket2] = await storage.createBucket(secondBucketName);
+  await bucket2.acl.add(bucketOptions);
 });
 
-test.after.always(async () => {
+after(async () => {
   tools.restoreConsole();
```
Comment (JustinBeckwith) [inline storage-transfer/system-test/transfer.test.js]: Same as above, we just dropped this

File: storage-transfer/system-test/transfer.test.js [hunk 4]
```diff
@@ -16,49 +16,46 @@
...
   const bucketOne = storage.bucket(firstBucketName);
   const bucketTwo = storage.bucket(secondBucketName);
   try {
```
Comment (JustinBeckwith) [inline storage-transfer/system-test/transfer.test.js]: Just for readability, could this be:

```js
// Attempt to do the delete twice, and swallow any errors
for (let i=0; i<2; i++) {
  await bucketOne.deleteFiles({force: true}).catch(console.warn);
}
```

File: storage-transfer/system-test/transfer.test.js [hunk 5]
```diff
@@ -85,65 +82,80 @@ test.cb.serial(`should create a storage transfer job`, t => {
...
-    t.is(transferJob.name.indexOf(`transferJobs/`), 0);
-    t.is(transferJob.description, description);
-    t.is(transferJob.status, `ENABLED`);
-    t.true(
-      console.log.calledWith(`Created transfer job: %s`, transferJob.name)
+    assert.strictEqual(transferJob.name.indexOf('transferJobs/'), 0);
+    assert.strictEqual(transferJob.description, description);
+    assert.strictEqual(transferJob.status, 'ENABLED');
+    assert.strictEqual(
+      console.log.calledWith('Created transfer job: %s', transferJob.name),
```
Comment (JustinBeckwith) [inline storage-transfer/system-test/transfer.test.js]: Ahhh, I see why the console.log stubbing now.

File: storage-transfer/system-test/transfer.test.js (hunk 4 above)
Comment (fhinkel) [reply storage-transfer/system-test/transfer.test.js]: Don't you dare adding `awaits`, that breaks everything. Trust me, we've been there.

File: storage-transfer/package.json (hunk 1 above)
Comment (fhinkel) [reply storage-transfer/package.json]: Happy to change it, I'll do it in a separate PR for all the folders.

**Comment by AVaksman** [comment]
towards #1080

**Pull request merged**